design-proposals/emts-scale-provisioning.md

```diff
@@ -0,0 +1,117 @@
+- Limited Web UI with EIM only.
+- No cluster and application orchestrator deployed.
+
+### PXE-based provisioning workflow
+
+For MVP we will heavily rely on capabilities provided by Tinkerbell SMEE. We will leverage DHCP and TFTP server implementations from SMEE and only make several modifications ot the 
+EIM stack.
+
+By default, Tinkerbell SMEE relies on MAC addresses to uniquely identify PXE-booting machines and customize the iPXE script per machine.
+In the case of EIM, we use a static iPXE script that is not customized per Edge Node. Therefore, we can avoid using MAC address as a unique identifier
+and it will let use avoid adding MAC address as another EN identifier during pre-registration.
+
+The below flow will be feasible to achieve with minimal modifications to the EIM stack. The following changes must be made:
+
+- Tinkerbell SMEE must be enabled via infra-charts to provide DHCP and TFTP server
+- Tinkerbell SMEE must be configured with the following flags to avoid lookups by MAC address and to use our EIM iPXE script:
+  - `-dhcp-mode=auto-proxy`
+  - `-dhcp-http-ipxe-script-url=<URL to iPXE script on Provisioning NGINX>` (e.g., `https://tinkerbell-nginx.CLUSTER_DOMAIN/tink-stack/boot.ipxe`)
+  - `-dhcp-http-ipxe-script-prepend-mac=false`
+- We may need to slightly adjust the `boot.ipxe` that is produced by DKAM to meet PXE boot requirements
+- We need to revise how we sign iPXE script and HookOS image - in the PXE-based provisioning flow we shouldn't upload any certificates/SB keys.
+- We may want to have a minimal Tinkerbell workflow for EMT-S (e.g., we don't need to install cloud-init since EMT-S nodes won't be connected to the same orchestrator instance, likely)
+
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+  box LightYellow Edge Node
+  participant bios as BIOS (PXE)
+  participant ipxe as iPXE
+  end
+  participant user as User
+  box rgb(235,255,255) Edge Orchestrator
+  participant smee as Tinkerbell SMEE
+  participant pa as Provisioning Nginx
+  participant inv as Inventory / API
+  end
+
+  user->>inv: Pre-register ENs with SN/UUID
+  note over bios,ipxe: PXE boot starts
+
+  bios->>smee: DHCP Discover
+  smee->>bios: DHCP reply with TFTP endpoint storing base EFI script
+
+  bios->>ipxe: Leaves PXE context, taken over by iPXE
+  ipxe->>smee: DHCP Discover
+  smee->>ipxe: DHCP reply with HTTP endpoint where boot.ipxe is stored
+
+  ipxe->>+pa: Download boot.ipxe
+  pa->>-ipxe: [boot.ipxe]
+
+  note over bios,inv: OS provisioning continues in a standard flow, through HookOS and Tinkerbell workflow
+```
+
+Notes:
+- Similar to the standard flow - If EN is not pre-registered, the process will fallback to Interactive Onboarding waiting for a user to provide credentials.
+- Similar to the standard flow - If EN is pre-registered, but the OS profile is not selected, the default OS will be provisioned.
```

Comment:
this is not true - it will be stuck in onboarded state. We removed the auto-provisioning

Reply:
@pierventre autoprovision is not removed, but only disabled. We could re-enable it, if we think it's useful for this use case. @krishnajs worth mentioning this, since by default it would be disabled.

Reply:
I know that is not removed but this is something that people did not like as UX and we disabled because People were not happy about.

Reply:
Yes, also we can achieve an equivalent workflow even with autoprovision disabled.